src/pybind/mgr/ansible/ansible_runner_svc.py

```diff
@@ -247,29 +246,58 @@ def http_post(self, endpoint, payload, params_dict = {}):
         :returns: A requests object
         """
 
-        response = None
-
         try:
             the_url = "%s/%s" % (self.server_url, endpoint)
-            r = requests.post(the_url,
+            response = requests.post(the_url,
                               verify = self.verify_server,
                               headers = {"Authorization": self.token,
                                          "Content-type": "application/json"},
                               json = payload,
                               params = params_dict)
 
-            if r.status_code != requests.codes.ok:
+            if response.status_code != requests.codes.ok:
                 self.log.error("http POST %s [%s] <--> (%s - %s:%s)\n",
-                              the_url, payload, r.status_code, r.reason, r.text)
+                              the_url, payload, response.status_code,
+                              response.reason, response.text)
             else:
                 self.log.info("http POST %s <--> (%s - %s)",
-                              the_url, r.status_code, r.text)
-            response = r
+                              the_url, response.status_code, response.text)
 
-        except Exception:
-            self.log.exception("Ansible runner service(POST %s)", the_url)
+            return response
+
+        except Exception as ex:
+            self.log.exception("Ansible runner service(POST %s): %s",
+                              (the_url, ex))
+            raise
+
+    def http_delete(self, endpoint):
+        """Execute an http delete request
+
+        :param endpoint: Ansible Runner service RESTful API endpoint
+
+        :returns: A requests object
+        """
+
+        try:
+            the_url = "%s/%s" % (self.server_url, endpoint)
+            response = requests.delete(the_url,
+                                verify = self.verify_server,
+                                headers = {"Authorization": self.token})
+
+            if response.status_code != requests.codes.ok:
+                self.log.error("http DELETE %s <--> (%s - %s)\n%s",
+                               the_url, response.status_code, response.reason,
+                               response.text)
+            else:
+                self.log.info("http DELETE %s <--> (%s - %s)",
+                              the_url, response.status_code, response.text)
```

Comment:
Just for my understanding: If the DELETE operation failed, we're just logging the error into the log file, without returning any error the the user?

Reply:
We should to differenciate two different cases.
1. You obtain a response, in this case the operation has been executed and we can have a succes or a fail, the upper layer will know what to do depending of the response status code.
2. No response, meand problem reaching the external orchestrator, an error is logged and the upper layer should know how to proceed. ( probably fill the Completion result with the error message)